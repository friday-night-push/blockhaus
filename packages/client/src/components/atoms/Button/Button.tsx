import { Button as ButtonComponent, ButtonProps } from '@gravity-ui/uikit';

export const Button = ({
  type = 'button',
  view = 'normal',
  children,
  ...props
}: ButtonProps) => {
  return (
    <ButtonComponent
      type={type}
      view={view}
      size="xl"
      pin={'brick-brick'}
      {...props}>
      {children}
    </ButtonComponent>
  );
};
