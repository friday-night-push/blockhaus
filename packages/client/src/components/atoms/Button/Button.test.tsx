import { render } from '@testing-library/react';

import { Button } from './Button';

describe('Button', () => {
  it('renders correctly with default props', () => {
    const { container } = render(<Button>Button</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
