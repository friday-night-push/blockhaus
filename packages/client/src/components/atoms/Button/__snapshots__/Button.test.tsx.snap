// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with default props 1`] = `
<button
  class="g-button g-button_view_flat g-button_size_xl g-button_pin_brick-brick"
  type="button"
>
  <span
    class="g-button__text"
  >
    Button
  </span>
</button>
`;
